### DISCUSSION TITLE
What would be a good approach mutating multiple caches on a shared item update?

### DISCUSSION THREAD

**Discussion by sveta-slepner** [TanStack/query#6563] [Q&A] [answered]
Let's say that i'm showing two lists of books in a page.
Something like:
```
const popularBooks = useQuery({
   queryKey: ['books', 'popular'],
   queryFn: ...
});

const saleBooks = useQuery({
   queryKey: ['books', 'sale'],
   queryFn: ...
})
```

And now I need to edit/remove a book that exists in both lists and mutate the data.
Is there a smart way to do it, other than knowing the exact cache keys of lists I want to edit?

Thanks!

**Comment by TkDodo** [comment, marked as answer]
`queryClient.setQueriesData(['books]', (prev => ...))` can do it

**Reply by sveta-slepner** [reply]
Thank you!
